### PR TITLE
 avfilter/x86/vf_exposure: add x86 SIMD optimization

### PR DESCRIPTION
**Pull request by iqsxw**
Performance data(Less is better):
exposure_c:    857394
exposure_asm:  327589

### CODE REVIEW

File: libavfilter/vf_exposure.c [hunk 1]
```diff
@@ -26,23 +26,20 @@
...
+static void exposure_c(float *ptr, int length, float black, float scale)
+{
+    int i;
 
-    float scale;
-    int (*do_slice)(AVFilterContext *s, void *arg,
-                    int jobnr, int nb_jobs);
-} ExposureContext;
+    for (i = 0; i < length; i++)
+            ptr[i] = (ptr[i] - black) * scale;
```
Comment (xuguangxin) [inline libavfilter/vf_exposure.c:36]: You can treat this as "ptr[i]  * scale - black * scale;"  and use _mm256_fmadd_ps

File: libavfilter/vf_exposure.c [hunk 2]
```diff
@@ -52,24 +49,27 @@ static int exposure_slice(AVFilterContext *ctx, void *arg, int jobnr, int nb_job
     for (int p = 0; p < 3; p++) {
         const int linesize = frame->linesize[p] / 4;
         float *ptr = (float *)frame->data[p] + slice_start * linesize;
-        for (int y = slice_start; y < slice_end; y++) {
-            for (int x = 0; x < width; x++)
-                ptr[x] = (ptr[x] - black) * scale;
-
-            ptr += linesize;
-        }
+        s->exposure_func(ptr, linesize * (slice_end - slice_start), black, scale);
```
Comment (xuguangxin) [inline libavfilter/vf_exposure.c:52]: linesize * 4 may > width.  you touch the useless pixels after width.

File: libavfilter/x86/vf_exposure.asm [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+%if WIN64
+    VBROADCASTSS m0, xmm2
+    VBROADCASTSS m1, xmm3
+%else
+    VBROADCASTSS m0, xmm0
+    VBROADCASTSS m1, xmm1
+%endif
+
+.loop:
+    movu        m2, [ptrq]
```
Comment (xuguangxin) [inline libavfilter/x86/vf_exposure.asm:49]: Unrolling your loop will speed up your program.

File: libavfilter/vf_exposure.c (hunk 1 above)
Comment (iqsxw) [reply libavfilter/vf_exposure.c:36]: Great! It did.

File: libavfilter/vf_exposure.c (hunk 2 above)
Comment (iqsxw) [reply libavfilter/vf_exposure.c:52]: It did touch the useless pixels but it's for alignment.

**Pull request closed without merging**